### PR TITLE
update golangci-lint to 1.49.0

### PR DESCRIPTION
**Pull request by anjannath**

### CODE REVIEW

**Comment by praveenkumar** [comment]
RHEL-8 still have 1.17.12 version of golang in official repo and also as part of go toolset images https://catalog.redhat.com/software/containers/ubi8/go-toolset/5ce8713aac3db925c03774d1

File: cmd/crc/cmd/daemon.go [hunk 1]
```diff
@@ -143,7 +143,12 @@ func run(configuration *types.Configuration) error {
 		mux := http.NewServeMux()
 		mux.Handle("/network/", http.StripPrefix("/network", vn.Mux()))
 		mux.Handle("/api/", http.StripPrefix("/api", api.NewMux(config, newMachine(), logging.Memory, segmentClient)))
-		if err := http.Serve(listener, handlers.LoggingHandler(os.Stderr, mux)); err != nil {
+		s := &http.Server{
+			Handler:      handlers.LoggingHandler(os.Stderr, mux),
+			ReadTimeout:  10 * time.Second,
+			WriteTimeout: 10 * time.Second,
+		}
+		if err := s.Serve(listener); err != nil {
```
Comment (cfergeau) [inline cmd/crc/cmd/daemon.go:151]: Fwiw, the commit fixing `make lint` could come before the commit updating golangci-lint to the latest version, this way `make check` would work for all commits in this PR. Not really important, but can be useful in projects using `git bisect` intensively.

**Review by cfergeau — COMMENTED** [review]
More discussion is needed around the go 1.18 commit, otherwise looks good to me!

**Comment by anjannath** [comment]
@praveenkumar @cfergeau dropped the golang version update commit

File: cmd/crc/cmd/daemon.go (hunk 1 above)
Comment (anjannath) [reply cmd/crc/cmd/daemon.go:151]: reordered the commits!

**Comment by anjannath** [comment]
seems like `golangci-lint` v1.49.0 has dependencies that needs go1.18 and go1.19, getting this error when go1.17 is installed:
```
$ make cross-lint
cd tools && GOBIN="/Users/anjan/work/crc-upstream-main/tools/bin" go install github.com/golangci/golangci-lint/cmd/golangci-lint
# golang.org/x/exp/constraints
vendor/golang.org/x/exp/constraints/constraints.go:13:2: syntax error: unexpected ~, expecting method or interface name
vendor/golang.org/x/exp/constraints/constraints.go:20:2: syntax error: unexpected ~, expecting method or interface name
vendor/golang.org/x/exp/constraints/constraints.go:27:9: syntax error: unexpected |, expecting semicolon or newline or }
vendor/golang.org/x/exp/constraints/constraints.go:34:2: syntax error: unexpected ~, expecting method or interface name
vendor/golang.org/x/exp/constraints/constraints.go:41:2: syntax error: unexpected ~, expecting method or interface name
vendor/golang.org/x/exp/constraints/constraints.go:49:10: syntax error: unexpected |, expecting semicolon or newline or }
note: module requires Go 1.18
# github.com/golangci/golangci-lint/pkg/golinters/goanalysis
vendor/github.com/golangci/golangci-lint/pkg/golinters/goanalysis/runner_loadingpackage.go:126:3: unknown field 'Instances' in struct literal of type types.Info
note: module requires Go 1.19
make: *** [/Users/anjan/work/crc-upstream-main/tools/bin/golangci-lint] Error 2
```

**Comment by cfergeau** [comment]
> seems like `golangci-lint` v1.49.0 has dependencies that needs go1.18 and go1.19, getting this error when go1.17 is installed:

Ah :( And I could not find a golang 1.18 in rhel8 or rhel9, so maybe it's better to delay this PR for a little while? :-/

**Pull request closed without merging**